coda-rag/py/Coda-RAG.ipynb

```diff
@@ -0,0 +1,539 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "vigvBmpNxHvb"
+   },
+   "source": [
+    "# Coda RAG Demo\n",
+    "\n",
+    "<a target=\"_blank\" href=\"https://colab.research.google.com/github/braintrustdata/braintrust-examples/blob/main/coda-rag/py/Coda-RAG.ipynb\">\n",
+    "  <img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/>\n",
+    "</a>\n",
+    "\n",
+    "Welcome to [BrainTrust](https://www.braintrustdata.com/)! In this notebook, you'll build and evaluate an AI app that answers questions about Coda's help desk.\n",
+    "\n",
+    "To provide the LLM with relevant information from Coda's help desk, we'll use a technique called RAG (retrieval-augmented generation) to infuse our prompts with text from the most-relevant sections of their docs. To evaluate the performance of our app, we'll use an LLM to generate question-answer pairs from the docs, and we'll use a technique called **model graded evaluation** to automatically evaluate the final responses against the expected answers.\n",
+    "\n",
+    "Before starting, please make sure that you have a BrainTrust account. If you do not, please [sign up](https://www.braintrustdata.com) or [get in touch](mailto:info@braintrustdata.com). After this tutorial, feel free to dig deeper by visiting [the docs](http://www.braintrustdata.com/docs)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "5AFngU_sw1K5"
+   },
+   "outputs": [],
+   "source": [
+    "# Set your API keys here\n",
+    "%env OPENAI_API_KEY=<YOUR_OPENAI_API_KEY>\n",
+    "%env BRAINTRUST_API_KEY=<YOUR_BRAINTRUST_API_KEY>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "rTIA4DVQw1K7",
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "%pip install -U autoevals braintrust requests openai lancedb markdownify"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "QA_GEN_MODEL = \"gpt-3.5-turbo\"\n",
+    "QA_ANSWER_MODEL = \"gpt-3.5-turbo\"\n",
+    "QA_GRADING_MODEL = \"gpt-4\"\n",
+    "RELEVANCE_MODEL = \"gpt-3.5-turbo\"\n",
+    "\n",
+    "NUM_SECTIONS = 20\n",
+    "NUM_QA_PAIRS = 20  # Increase this number to test and fine-tune at a larger scale"
```

Comment:
maybe remove reference to fine-tune since the notebook doesn't (yet) do it?

Reply:
oops